File: exercice2/scala/src/test/scala/fr/devoxx/devops/logs/sql/SparkSQL2Test.scala
package fr.devoxx.devops.logs.sql

import fr.devoxx.devops.logs.{SharedSparkSQLContext, SharedSparkContext}
import org.scalatest.{FunSuite, Matchers}

class SparkSQL2Test extends FunSuite with SharedSparkSQLContext with Matchers {

  test("Répartition des codes http") {
    val result = SparkSQL2(rdd, sqlContext).process
    result shouldBe (Array((200, 6067), (404, 35)))
  }
}

File: exercice2/java/src/test/java/fr/devoxx/devops/logs/sql/SparkSQL2Test.java
package fr.devoxx.devops.logs.sql;

import fr.devoxx.devops.logs.SparkTest;
import org.junit.Test;
import scala.Tuple2;

import java.util.List;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.contains;
import static org.hamcrest.Matchers.containsInAnyOrder;
import static org.hamcrest.Matchers.hasSize;

public class SparkSQL2Test extends SparkTest {
    @Test
    public void test_spark() {
        SparkSQL2 test = new SparkSQL2();
        List<Tuple2<Integer, Long>> result = test.process(rdd(), sqlContext());
        assertThat(result, hasSize(2));
        assertThat(result, contains(new Tuple2<>(200, 6067L), new Tuple2<>(404, 35L)));
    }

}
